pkg/master/controller.go

```diff
@@ -155,15 +174,14 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
 			Labels:    map[string]string{"provider": "kubernetes", "component": "apiserver"},
 		},
 		Spec: api.ServiceSpec{
-			Ports: []api.ServicePort{{Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}},
+			Ports: ports,
 			// maintained by this code, not by the pod selector
 			Selector:        nil,
 			ClusterIP:       serviceIP.String(),
```

Comment:
- I think a simplified problem statement would be "Cross-platform option for users outside of the cluster to leverage the  introspective API".  
- I also think you're point, which is that "the kubelet is a weird example, since kubelets, by definition, can leverage the introspective API", is totally valid.